Order Book streaming doesn't work

```kotlin
var server = Server("https://horizon.stellar.org")
var book = server.orderBook()
        .buyingAsset(Asset.create("crypto", "BTC", "GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH"))
        .sellingAsset(Asset.create("native", null, null))
        .stream({ book ->
        println(book.base)
    })
```

```
Exception in thread "OkHttp Dispatcher" java.lang.IllegalStateException: Failed 200
	at org.stellar.sdk.requests.SSEStream$StellarEventSourceListener.onFailure(SSEStream.java:176)
	at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:83)
	at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54)
	at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
	at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.UnsupportedOperationException: this response does not have a paging token
	at org.stellar.sdk.responses.Response.getPagingToken(Response.java:34)
	at org.stellar.sdk.requests.SSEStream$StellarEventSourceListener.onEvent(SSEStream.java:189)
	at shadow.okhttp3.internal.sse.RealEventSource.onEvent(RealEventSource.java:111)
	at shadow.okhttp3.internal.sse.ServerSentEventReader.completeEvent(ServerSentEventReader.java:112)
	at shadow.okhttp3.internal.sse.ServerSentEventReader.processNextEvent(ServerSentEventReader.java:69)
	at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:80)
	... 6 more
```

--- a/src/main/java/org/stellar/sdk/requests/SSEStream.java
+++ b/src/main/java/org/stellar/sdk/requests/SSEStream.java
@@ -9,6 +9,7 @@ import okhttp3.sse.EventSource;
 import okhttp3.sse.EventSourceListener;
 import org.stellar.sdk.Util;
 import org.stellar.sdk.responses.GsonSingleton;
+import org.stellar.sdk.responses.Pageable;
 
 import javax.annotation.Nullable;
 import java.io.Closeable;
@@ -195,8 +196,10 @@ public class SSEStream<T extends org.stellar.sdk.responses.Response> implements 
         return;
       }
       T event = GsonSingleton.getInstance().fromJson(data, responseClass);
-      String pagingToken = event.getPagingToken();
-      requestBuilder.cursor(pagingToken);
+      if (event instanceof Pageable){
+        String pagingToken = ((Pageable)event).getPagingToken();
+        requestBuilder.cursor(pagingToken);
+      }
       stream.lastEventId.set(id);
       listener.onEvent(event);
     }

--- a/src/main/java/org/stellar/sdk/responses/AccountResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/AccountResponse.java
@@ -22,8 +22,6 @@ public class AccountResponse extends Response implements org.stellar.sdk.Transac
   private KeyPair keypair;
   @SerializedName("sequence")
   private Long sequenceNumber;
-  @SerializedName("paging_token")
-  private String pagingToken;
   @SerializedName("subentry_count")
   private Integer subentryCount;
   @SerializedName("inflation_destination")
@@ -71,11 +69,7 @@ public class AccountResponse extends Response implements org.stellar.sdk.Transac
   public void incrementSequenceNumber() {
     sequenceNumber++;
   }
-
-  public String getPagingToken() {
-    return pagingToken;
-  }
-
+  
   public Integer getSubentryCount() {
     return subentryCount;
   }

--- a/src/main/java/org/stellar/sdk/responses/AssetResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/AssetResponse.java
@@ -3,7 +3,7 @@ package org.stellar.sdk.responses;
 import com.google.gson.annotations.SerializedName;
 import org.stellar.sdk.Asset;
 
-public class AssetResponse extends Response {
+public class AssetResponse extends Response implements Pageable {
     @SerializedName("asset_type")
     private final String assetType;
     @SerializedName("asset_code")

--- a/src/main/java/org/stellar/sdk/responses/LedgerResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/LedgerResponse.java
@@ -8,7 +8,7 @@ import com.google.gson.annotations.SerializedName;
  * @see org.stellar.sdk.requests.LedgersRequestBuilder
  * @see org.stellar.sdk.Server#ledgers()
  */
-public class LedgerResponse extends Response {
+public class LedgerResponse extends Response implements Pageable {
   @SerializedName("sequence")
   private final Long sequence;
   @SerializedName("hash")

--- a/src/main/java/org/stellar/sdk/responses/OfferResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/OfferResponse.java
@@ -11,7 +11,7 @@ import org.stellar.sdk.KeyPair;
  * @see org.stellar.sdk.requests.OffersRequestBuilder
  * @see org.stellar.sdk.Server#offers()
  */
-public class OfferResponse extends Response {
+public class OfferResponse extends Response implements Pageable {
   @SerializedName("id")
   private final Long id;
   @SerializedName("paging_token")

--- a/src/main/java/org/stellar/sdk/responses/Pageable.java
+++ b/src/main/java/org/stellar/sdk/responses/Pageable.java
@@ -1,0 +1,5 @@
+package org.stellar.sdk.responses;
+
+public interface Pageable {
+    public String getPagingToken();
+}

--- a/src/main/java/org/stellar/sdk/responses/Response.java
+++ b/src/main/java/org/stellar/sdk/responses/Response.java
@@ -29,11 +29,6 @@ public abstract class Response {
     return rateLimitLimit;
   }
 
-
-  public String getPagingToken() {
-    throw new UnsupportedOperationException("this response does not have a paging token");
-  }
-
   /**
    * Returns X-RateLimit-Remaining header from the response.
    * The number of remaining requests for the current window.

--- a/src/main/java/org/stellar/sdk/responses/TradeResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/TradeResponse.java
@@ -12,7 +12,7 @@ import org.stellar.sdk.Price;
  * @see org.stellar.sdk.requests.TradesRequestBuilder
  * @see org.stellar.sdk.Server#trades()
  */
-public class TradeResponse extends Response {
+public class TradeResponse extends Response implements Pageable {
     @SerializedName("id")
     private final String id;
     @SerializedName("paging_token")

--- a/src/main/java/org/stellar/sdk/responses/TransactionResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/TransactionResponse.java
@@ -13,7 +13,7 @@ import static com.google.common.base.Preconditions.checkNotNull;
  * @see org.stellar.sdk.requests.TransactionsRequestBuilder
  * @see org.stellar.sdk.Server#transactions()
  */
-public class TransactionResponse extends Response {
+public class TransactionResponse extends Response implements Pageable {
   @SerializedName("hash")
   private final String hash;
   @SerializedName("ledger")

--- a/src/main/java/org/stellar/sdk/responses/effects/EffectResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/effects/EffectResponse.java
@@ -4,6 +4,7 @@ import com.google.gson.annotations.SerializedName;
 
 import org.stellar.sdk.KeyPair;
 import org.stellar.sdk.responses.Link;
+import org.stellar.sdk.responses.Pageable;
 import org.stellar.sdk.responses.Response;
 
 /**
@@ -12,7 +13,7 @@ import org.stellar.sdk.responses.Response;
  * @see org.stellar.sdk.requests.EffectsRequestBuilder
  * @see org.stellar.sdk.Server#effects()
  */
-public abstract class EffectResponse extends Response {
+public abstract class EffectResponse extends Response implements Pageable {
   @SerializedName("id")
   protected String id;
   @SerializedName("account")

--- a/src/main/java/org/stellar/sdk/responses/operations/OperationResponse.java
+++ b/src/main/java/org/stellar/sdk/responses/operations/OperationResponse.java
@@ -4,6 +4,7 @@ import com.google.gson.annotations.SerializedName;
 
 import org.stellar.sdk.KeyPair;
 import org.stellar.sdk.responses.Link;
+import org.stellar.sdk.responses.Pageable;
 import org.stellar.sdk.responses.Response;
 
 /**
@@ -12,7 +13,7 @@ import org.stellar.sdk.responses.Response;
  * @see org.stellar.sdk.requests.OperationsRequestBuilder
  * @see org.stellar.sdk.Server#operations()
  */
-public abstract class OperationResponse extends Response {
+public abstract class OperationResponse extends Response implements Pageable {
   @SerializedName("id")
   protected Long id;
   @SerializedName("source_account")

--- a/src/test/java/org/stellar/sdk/responses/AccountDeserializerTest.java
+++ b/src/test/java/org/stellar/sdk/responses/AccountDeserializerTest.java
@@ -11,7 +11,6 @@ public class AccountDeserializerTest extends TestCase {
   public void testDeserialize() {
     AccountResponse account = GsonSingleton.getInstance().fromJson(json, AccountResponse.class);
     assertEquals(account.getKeypair().getAccountId(), "GAAZI4TCR3TY5OJHCTJC2A4QSY6CJWJH5IAJTGKIN2ER7LBNVKOCCWN7");
-    assertEquals(account.getPagingToken(), "1");
     assertEquals(account.getSequenceNumber(), new Long(2319149195853854L));
     assertEquals(account.getSubentryCount(), new Integer(0));
     assertEquals(account.getInflationDestination(), "GAGRSA6QNQJN2OQYCBNQGMFLO4QLZFNEHIFXOMTQVSUTWVTWT66TOFSC");

--- a/src/test/java/org/stellar/sdk/responses/AccountsPageDeserializerTest.java
+++ b/src/test/java/org/stellar/sdk/responses/AccountsPageDeserializerTest.java
@@ -12,7 +12,6 @@ public class AccountsPageDeserializerTest extends TestCase {
     Page<AccountResponse> accountsPage = GsonSingleton.getInstance().fromJson(json, new TypeToken<Page<AccountResponse>>() {}.getType());
 
     assertEquals(accountsPage.getRecords().get(0).getKeypair().getAccountId(), "GAAZI4TCR3TY5OJHCTJC2A4QSY6CJWJH5IAJTGKIN2ER7LBNVKOCCWN7");
-    assertEquals(accountsPage.getRecords().get(0).getPagingToken(), "1");
     assertEquals(accountsPage.getRecords().get(9).getKeypair().getAccountId(), "GACFGMEV7A5H44O3K4EN6GRQ4SA543YJBZTKGNKPEMEQEAJFO4Q7ENG6");
     assertEquals(accountsPage.getLinks().getNext().getHref(), "/accounts?order=asc&limit=10&cursor=86861418598401");
     assertEquals(accountsPage.getLinks().getPrev().getHref(), "/accounts?order=desc&limit=10&cursor=1");

